src/mlpack/methods/ann/visitor/reset_visitor.hpp

```diff
@@ -33,6 +33,22 @@ class ResetVisitor : public boost::static_visitor<void>
   void operator()(MoreTypes layer) const;
 
  private:
+  //! Execute the ResetCell() function for a layer model which implements
+  //! the Model() function.
+  template<typename T>
+  typename std::enable_if<
+      HasResetCheck<T, std::vector<LayerTypes>&(T::*)()>::value && 
+      !HasModelCheck<T>::value,void>::type
```

Comment:
Hey @zoq I am trying to add a function for reset cell model in resetvisitor because we are going to use ffn as a layer can you  suggest a right way to check model.

Reply:
if you add a construct like this you have to make sure you implement each case, so: first expression true, second false, first false, second true, both false, both true. Using the same expressions for each case. Hopefully that is helpful, let me know if I should clarify anything.

Reply:
Ok Thanks for clarification one more I don't know why I am getting error like this 
```
/home/vsts/work/1/s/src/mlpack/methods/ann/layer/../visitor/reset_visitor.hpp:40:46: error: type/value mismatch at argument 1 in template parameter list for ‘template<class _Tp, class _Alloc> class std::vector’
       HasResetCheck<T, std::vector<LayerTypes>&(T::*)()>::value && 

```
Is it because of LayerTypes

Reply:
Looks like you solved the issue?

Reply:
Yes I solved that issue but this method was not working. So, I used a simple hack to reset parameter I am trying to add tests but they are failing I don't know why may be there is something wrong with implementation I will inform you @zoq if I find something wrong.